lib/gollum-lib/wiki.rb

```diff
@@ -431,7 +440,7 @@ def rename_page(page, rename, commit = {})
     # Returns the String SHA1 of the newly written version, or the
     # Gollum::Committer instance if this is part of a batch update.
     def update_page(page, name, format, data, commit = {})
-      name     ||= page.name
+      name     = name.present? ? ::File.basename(name) : page.name
```

Comment:
If the method update_page can only rename files, let's stay strict to that and disallow moving the file in the current directory tree. If we do this, we don't have to deal with other problems in the rename_page operation.